File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.java
class Solution {
    public int longestSubarray(int[] nums) {
        int max = 0, count = 0,temp_count = 0, len = nums.length;
        boolean cheak = false;

        for(int i = 0; i < len; i++) {
            if(nums[i] > max) {
                max = nums[i];
                count = 1;
                cheak = true;
            }
            else if(cheak && nums[i] == max) {
                count++;
            }
            else if(!cheak && nums[i] == max) {
                temp_count++;
            }
            else {
                count = Math.max(count, temp_count);
                cheak = false;
                temp_count = 0;
            }
        }

        return count;
    }
}

File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        length = len(nums)
        max_val = 0
        max_answer = 0
        cheak = False

        for i in range(length) :
            if max_val <= nums[i] :
                cheak = True
            if max_val < nums[i] :
                max_val = nums[i]
                max_answer = 0
                cur = 1
            elif cheak & (nums[i] == max_val):
                cur += 1
            else :
                cur = 0
                cheak = False
            
            max_answer = max(max_answer, cur)

        return max_answer
            
